acme/setup.py

```diff
@@ -21,6 +21,7 @@
     'requests[security]>=2.4.1',  # security extras added in 2.4.1
     'setuptools',
     'six>=1.9.0',  # needed for python_2_unicode_compatible
+    'typing',
```

Comment:
>  Figure out if we should depend on this conditionally or not.
I don't think see a need to do that. `typing` is also available in Python3.

Wait a second. We install these with `pip`. What does whether or not it's packaged by distros have to do with anything?

Reply:
Whether or not `typing` is packaged by distros is relevant because when people install Certbot through their OS package manager (which is still the recommended way to get Certbot on most systems), `pip` isn't used and unfortunately `typing` isn't widely packaged.

I agree there's a strong argument that playing with conditional dependencies might not be worth it. I think the biggest appeal is that systems that are using Python 3.5+ don't need to install or package `typing` when it's a noop package which could make some of this easier for us.